### DISCUSSION TITLE
How to Extract and Plot Elastic Strain Data Along a Specified Path？

### DISCUSSION THREAD

**Discussion by kight55** [neperfepx/FEPX#124] [Q&A] [answered]
I would like to extract the elastic strain data along a specific path (x=0.5,y=0.5) and plot the distribution of elastic strain along the path (as shown in Figure b). Is there an appropriate method to extract the elastic strain values along the path?
![QQ20241206-170732](https://github.com/user-attachments/assets/18e57161-3bc5-47ff-9b03-26e040484b13)

**Comment by rquey** [comment, marked as answer]
There is no direct way (in Neper or FEPX) at the moment.

However, with Neper, you can do it by creating a [position file](https://neper.info/doc/fileformat.html#position-file) with the position of points along the line segment, loading that position file in -M and extracting the element to which each point belongs:
```
neper -M myfile.tess -loadmesh myfile.msh -loadpoint mypositionfile.txt -statpoint elt
```
To work on a deformed mesh, you can use [`-transform "node(<file_name>)"`](https://neper.info/doc/neper_m.html#cmdoption-transform).

Then, from the point positions and the elemental values (of the elastic strain), you can easily generate the profile.

**Reply by kight55** [reply]
Thank you very much, the answer is very helpful!
